protocols/raft/src/main/java/io/atomix/protocols/raft/session/impl/RaftSessionContext.java

```diff
@@ -405,6 +459,7 @@ public void registerIndexQuery(long index, Runnable query) {
    * @param result   The result.
    */
   public void registerResult(long sequence, OperationResult result) {
+    setRequestSequence(sequence);
```

Comment:
Setting the request sequence number after a command is applied helps reduce the amount of time it takes for a new leader leader to be able to accept new commands for a given session. Once the last written command is applied it can begin writing new commands in sequential order.